### PR TITLE
add Binaryen(Get|Set)AllowHeavyweight to binaryen-c.h

### PR DESCRIPTION
**Pull request by MaxGraey**
It's finished #3032 and fix https://github.com/AssemblyScript/assemblyscript/pull/1449

### CODE REVIEW

File: src/binaryen-c.h [hunk 1]
```diff
@@ -2124,6 +2124,12 @@ BINARYEN_API BinaryenIndex BinaryenGetOneCallerInlineMaxSize(void);
 // Applies to all modules, globally.
 BINARYEN_API void BinaryenSetOneCallerInlineMaxSize(BinaryenIndex size);
 
+// Gets the currently set of inlining heavyweight functions.
```
Comment (dcodeIO) [inline src/binaryen-c.h:2127]:
```suggestion
// Gets whether heavyweight functions are allowed to be inlined.
// Applies to all modules, globally.
```
Does that describe it accurately?

File: src/binaryen-c.h [hunk 2]
```diff
@@ -2124,6 +2124,12 @@ BINARYEN_API BinaryenIndex BinaryenGetOneCallerInlineMaxSize(void);
...
+BINARYEN_API int BinaryenGetAllowHeavyweight(void);
+
+// Enables or disables inlining heavyweight functions.
```
Comment (dcodeIO) [inline src/binaryen-c.h:2130]:
```suggestion
// Sets whether heavyweight functions are allowed to be inlined.
// Applies to all modules, globally.
```

**Comment by dcodeIO** [comment]
Looks good, this will fix the issue encountered on our end. Wondering though if we should add tests at least calling allowHeavyweight and the other (rather new) optimization options like inlining sizes to `test/binaryen.js/kitchen-sink.js`, to be sure.

**Review by dcodeIO — APPROVED** [review]

**Comment by dcodeIO** [comment]
Took the liberty to merge this one as it's rather straight forward and only affecting parts of Binaryen that I'm relatively familiar with. Let me know if you'd prefer to be involved in review of trivial fixes like these instead :)

**Comment by kripken** [comment]
@dcodeIO :+1: I think it's great to review and merge straightforward stuff like this as you did!

And thanks @MaxGraey for the PR!

**Comment by MaxGraey** [comment]
@kripken Thanks you! Btw I also found sometimes we could got compiler's halt problem when try inline recurcive functions, so I created another PR which fix that: https://github.com/WebAssembly/binaryen/pull/3085

**Pull request merged**